orangeduck/Cello
# CMake Support + Unix EndLine

## CMake Support

I have written CMakeLists.txt files to: 
- build Cello library, tests and demos
- install Cello library and associated headers
- check platform 
- specify compilation depending on:
  - release mode (debug/release)
  - build platform (Clang/GCC/MSVC)

I manage to build Cello with:
- GNU Makefiles on Ubuntu
- QtCreator on Ubuntu
- KDevelop3 on Ubuntu
- GNU Makefiles on OSX
- XCode on OSX

I added tests for MSVC compiler, but I don't manage to build library with this compiler. MSVC doesn't support C99 and doesn't understand most of the macros in Prelude.h
I targeted to use Cello to develop application that should run onto (INtime Real Time OS) and TwinCAT that both use MSVC as compiler. But it doesn't seem to fit my needs as it seems hard to build the library with this compiler
## Headers location

I move headers to include/Cello. This way they are organised as they will be installed => we don't need to install Cello on our system to test it and launch tests/demos.

I update #include in all implementation files to this new location.
## Unix EndLine

I don't make any change to the source code except that I change EndLine parameters to UNIX. I found both Windows and Unix EndLine usage in source code.
## Conclusions

Hope CMake support will still help you

Bertrand
